src/components/network-map-viewer/network/network-map.jsx

```diff
@@ -658,6 +781,52 @@ NetworkMap.propTypes = {
     onSubstationClickChooseVoltageLevel: PropTypes.func,
     onSubstationMenuClick: PropTypes.func,
     onVoltageLevelMenuClick: PropTypes.func,
+    onDrawPolygonModeActive: PropTypes.func,
+    onPolygonChanged: PropTypes.func,
 };
 
 export default React.memo(NetworkMap);
+
+function getSubstationsInPolygon(features, mapEquipments, geoData) {
+    const firstPolygonFeatures = Object.values(features)[0];
+    const polygonCoordinates = firstPolygonFeatures?.geometry;
+    if (!polygonCoordinates || polygonCoordinates.coordinates < 3) {
+        return [];
+    }
+    //get the list of substation
+    const substationsList = mapEquipments?.substations ?? [];
+    //for each substation, check if it is in the polygon
+    return substationsList // keep only the sybstation in the polygon
+        .filter((substation) => {
+            const pos = geoData.getSubstationPosition(substation.id);
+            return booleanPointInPolygon(pos, polygonCoordinates);
+        });
+}
+
+function getSelectedLinesInPolygon(
+    network,
+    lines,
+    geoData,
+    polygonCoordinates
+) {
+    return lines.filter((line) => {
+        try {
+            const linePos = geoData.getLinePositions(network, line);
+            if (!linePos) {
+                return false;
+            }
+            if (linePos.length < 2) {
+                return false;
+            }
+            //TODO : in the future, we could check if the line is fully in the polygon ( instead of just one point)
+            // we need to have a boolean that checks if detailed line is displayed
+            const linesExtremity = [linePos[0], linePos[linePos.length - 1]];
```

Comment:
Only when the lines are drawing without detailled positions !

Reply:
done,